lib/ash_paper_trail.ex

```diff
@@ -13,21 +13,36 @@ defmodule AshPaperTrail do
 
   @regex ~r/\.Version$/
   def allow_resource_versions(nil, resource) do
-    resource_name = to_string(resource)
-
-    if String.match?(resource_name, @regex) do
-      original_resource =
-        try do
-          resource_name
-          |> String.replace(@regex, "")
-          |> String.to_existing_atom()
-        rescue
-          ArgumentError -> false
+    cond do
+      is_atom(resource) and function_exported?(resource, :resource_version?, 0) and
```

Comment:
I think we may need to `Code.ensure_loaded?(resource)` here to ensure the module is loaded.